web/course.go

```diff
@@ -94,15 +96,81 @@ func CoursePage(c *gin.Context) {
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
+
+	progs, err := dao.LoadProgressesForCourseAndUser((*tumLiveContext.Course).ID, (*tumLiveContext.User).ID)
+	if err != nil {
+		sentry.CaptureException(err)
+		log.Printf("%v", err)
+	}
+
+	progressResponses, progressStreams := prepareCourseProgressData(tumLiveContext, progs)
+
+	encoded, err := json.Marshal(progressResponses)
+	if err != nil {
+		sentry.CaptureException(err)
+		log.Printf("%v", err)
+	}
 	// in any other case assume either validated before or public course
-	err := templ.ExecuteTemplate(c.Writer, "course.gohtml", CoursePageData{IndexData: indexData, Course: *tumLiveContext.Course})
+	err = templ.ExecuteTemplate(c.Writer, "course-overview.gohtml",
+		CoursePageData{IndexData: indexData, Course: *tumLiveContext.Course, StreamsWithProgress: progressStreams, ProgressResponses: string(encoded)})
 	if err != nil {
 		sentrygin.GetHubFromContext(c).CaptureException(err)
 	}
 }
 
+func prepareCourseProgressData(tumLiveContext tools.TUMLiveContext, streamProgresses []model.StreamProgress) ([]ProgressInfo, []ProgressStream) {
+	var progressStreams []ProgressStream
+	var progressResponses []ProgressInfo
```

Comment:
Can we use `make(...)` to allocate the required memory beforehand?

Reply:
We would have calculate the number of recordings then. But that would work. :+1:

Reply:
I didn't use make as most lectures only have like 20 streams maximum. I don't think really think it's needed here and has the potential to cause problems with my implementation.